languages/move_on_aptos/generic/Parse_move_on_aptos_tree_sitter.ml

```diff
@@ -1498,33 +1699,40 @@ and map_deep_ellipsis (env : env) ((v1, v2, v3) : CST.deep_ellipsis) =
   let v3 = (* "...>" *) token env v3 in
   G.DeepEllipsis (v1, v2, v3) |> G.e
 
+and map_identifier_or_anon_field (env : env)
+    ?(type_args : type_arguments option = None)
+    (x : CST.identifier_or_anon_field) =
+  match x with
+  | `Id tok ->
+      let ident = (* identifier *) str env tok in
+      G.FN (H2.add_type_args_opt_to_name (H2.name_of_id ident) type_args)
+  | `Anon_field_index tok -> (
+      (* pattern \d+ *)
+      let s, t = str env tok in
+      let literal = G.L (G.Int (Parsed_int.parse (s, t))) |> G.e in
+      match type_args with
+      | Some _ -> failwith "Type arguments are not supported for anon fields"
```

Comment:
I'd prefer to see some attempt at constructing a reasonable AST rather than refusing to parse the entire file if this is encountered (and even if it's not permitted in the grammar currently, I've seen cases where the grammar is updated and we start hitting things like this, or if we get a partial parse from tree sitter after encountering a type error)

Reply:
Fixed by translating the positional field with type args as a name (very awkward of course)